docs/learn.md

```diff
@@ -721,20 +723,1250 @@ template](https://github.com/melange-re/melange-opam-template).
+Js.Nullable.t | `null` / `undefined`
+Option.t `None` | `undefined`
+Option.t `Some( Some .. Some (None))` | internal representation
+Option.t `Some 2` | `2`
+record `{x: 1; y: 2}` | object `{x: 1, y: 2}`
+int64 | array of 2 elements `[high, low]` high is signed, low unsigned
+char | `'a'` -> `97` (ascii code)
+bytes | number array
+list `[]` | `0`
+list `[x, y]` | `[x, [y, 0]]`
+list `[1, 2, 3]` | `[ 1, [ 2, [ 3, 0 ] ] ]`
+variant | See below
+polymorphic variant | See below
+
+Variants with a single non-nullary constructor:
+
+```ocaml
+type tree = Leaf | Node of int * tree * tree
+(* Leaf -> 0 *)
+(* Node(7, Leaf, Leaf) -> { _0: 7, _1: 0, _2: 0 } *)
+```
+
+Variants with more than one non-nullary constructor:
+
+```ocaml
+type t = A of string | B of int
+(* A("foo") -> { TAG: 0, _0: "Foo" } *)
+(* B(2) -> { TAG: 1, _0: 2 } *)
+```
+
+Polymorphic Variant:
+
+```ocaml
+let u = `Foo (* "Foo" *)
+let v = `Foo(2) (* { NAME: "Foo", VAL: "2" } *)
+```
+
+> **_NOTE:_** Relying on runtime representations by reading or writing them
+manually from JavaScript code that communicates with Melange code might lead to
+runtime errors, as these representations might be changed in the future. In
+general, it is safer to leave the Melange compiler deal with the manipulation of
+these runtime values and interact with them from Melange code using bindings.
+
+Let’s see now some of these types in detail.
+
+### Shared types
+
+The following are types that can be shared between Melange and JavaScript almost
+"as is". Specific caveats are mentioned on the sections where they apply.
+
+#### Strings
+
+Melange strings are an immutable sequence of bytes. This is problematic when
+interacting with JavaScript code, because if one tries to use some unicode
+characters, like:
+
+```ocaml
+let () = Js.log "你好"
+```
+
+It will lead to some cryptic console output. To rectify this, Melange allows to
+define [quoted string
+literals](https://v2.ocaml.org/manual/lex.html#sss:stringliterals) using the
+`js` identifier, for example:
+
+```ocaml
+let () = Js.log {js|你好，
+世界|js}
+```
+
+For convenience, Melange exposes another special quoted string identifier: `j`.
+It is similar to JavaScript’ string interpolation, but for variables only (not
+arbitrary expressions):
+
+```ocaml
+let world = {j|世界|j}
+let helloWorld = {j|你好，$world|j}
+```
+
+You can surround the interpolation variable in parentheses too: `{j|你
+好，$(world)|j}`.
+
+To work with strings, Melange standard library provides some utilities in the
+[`Stdlib.String` module](todo-fix-me.md). The bindings to the native JavaScript
+functions to work with strings are in the [`Js.String` module](todo-fix-me.md).
+
+#### Float
+
+Melange floats are JavaScript numbers, and vice-versa. The Melange standard
+library doesn’t come with a Float module. The JavaScript Float API can be found
+in the [`Js.Float`](todo-fix-me.md) module.
+
+#### Int
+
+**Ints are 32-bits**. They compile to JavaScript numbers but treating both as
```

Comment:
I tried to adapt the text, but then a few lines below we mention:

> treating both as the same might lead to unexpected behavior, because of the differences in precision

as JS ints might get up to 2^53 - 1. So we would need to explain how JavaScript can adapt the integers depth dynamically based on the operations applied to them.

I fear it'd be too much detail to include these JS details for the more generic / entry level manual to Melange. Wdyt?

Reply:
Actually, I changed my mind and ended up adding explanation about all this in https://github.com/melange-re/documentation-site/pull/12/commits/e6c23fc2ea7f4b223899f2ea648d13b2fb2e39eb. 🤷‍♂️

Reply:
> integers themselves have a [max depth of 53

this is wrong, after 2^32-1 they're floating point.

Reply:
the rest looks good though.

Reply:
Gave another try in https://github.com/melange-re/documentation-site/pull/12/commits/1484685819554d08386a6ab357c482e7265bfee7, lmk what you think.

Reply:
And https://github.com/melange-re/documentation-site/pull/12/commits/cc4423bfe041f07eddd8b9b0b7448c13b506296c.